test/src/gatherers/critical-network-chains.js

```diff
@@ -0,0 +1,118 @@
+/**
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const GathererClass = require('../../../src/gatherers/critical-network-chains');
+const assert = require('assert');
+
+const Gatherer = new GathererClass();
+
+function mockTracingData(prioritiesList, edges) {
+  const networkRecords = prioritiesList.map((priority, index) =>
+      ({requestId: index,
+        initialPriority: () => priority,
+        initiatorRequest: () => null}));
+
+  // add mock initiator information
+  edges.forEach(edge => {
+    networkRecords[edge[1]].initiatorRequest = () => ({requestId: edge[0]});
+  });
+
+  return networkRecords;
+}
+
+function testGetCriticalChain(data) {
+  const mockNetworkRecords = mockTracingData(data.priorityList, data.edges);
+  const criticalChains = Gatherer.getCriticalChains(mockNetworkRecords);
+  // It is sufficient to only check the requestIds are correct in the chain
+  const requestIdChains = criticalChains.map(chain =>
+    chain.map(node => node.requestId));
+  // Ordering of the chains do not matter
+  assert.deepEqual(new Set(requestIdChains), new Set(data.expectedChains));
```

Comment:
Wow this is a bummer. Spreading them won't work because it doesn't get rid of the ordering issue. I might as well compare the lists
```javascript
const a = new Set([1, 2, 3]);
const b = new Set([1, 3, 2]);
[...a.values()]  // [1, 2, 3]
[...b.values()]  // [1, 3, 2]
```